src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -966,6 +962,15 @@ private void joinTokenRing(int delay, boolean autoBootstrap, Collection<String>
                 setMode(Mode.JOINING, "waiting for schema information to complete", true);
                 Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
             }
+
+            while(!isSchemaConsistent())
```

Comment:
i don't see a reason not to just combine this with the previous check and have it be

```
            while (!MigrationManager.isReadyForBootstrap() || !isSchemaConsistent())
            {
                setMode(Mode.JOINING, "waiting for schema information to complete... local schema version is {Schema.instance.getVersion().toString()}", true);
                Uninterruptibles.sleepUninterruptibly(1, TimeUnit.SECONDS);
            }
```

Reply:
this also means we can remove the above `// first sleep the delay to make sure we see all our peers`, right?